### PR TITLE
Add depends on field to the CR

### PR DESCRIPTION
**Pull request by whites11**
Towards: https://github.com/giantswarm/giantswarm/issues/25504

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Review by paurosello — APPROVED** [review]
LGTM, is the status removal expexted?

File: config/crd/release.giantswarm.io_releases.yaml [hunk 1]
```diff
@@ -164,9 +169,3 @@ spec:
     storage: true
     subresources:
       status: {}
-status:
-  acceptedNames:
-    kind: ""
-    plural: ""
-  conditions: []
-  storedVersions: []
```
Comment (njuettner) [inline config/crd/release.giantswarm.io_releases.yaml:172]: Why is this removed?

Comment (whites11) [reply config/crd/release.giantswarm.io_releases.yaml:172]: this was done by `controller-gen` so I don't know why that's the case.

Comment (njuettner) [reply config/crd/release.giantswarm.io_releases.yaml:172]: hmm ok, let's merge it 👍🏻

**Review by njuettner — APPROVED** [review]
